optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/uni/UniConstraintStream.java

```diff
@@ -186,20 +187,109 @@
     // Group by
     // ************************************************************************
 
-    <GroupKey_, ResultContainer_, Result_> UniConstraintStream<Result_> groupBy(
+    /**
+     * Runs all items in the stream through a given @{@link UniConstraintCollector} and converts them into a new
+     * {@link UniConstraintStream} which only has one item in it, the result of applying
+     * {@link UniConstraintCollector#finisher()}.
```

Comment:
How we figure out a good example format in a bluejeans meeting and add an example in every javadoc?

I am thinking something like:

```
[expanation]
<p>
For example, a Uni<Person> with elements [Ann(age 100), Beth (age 20), Carl (age 3)] groupBy sum(Person::getAge) returns a Uni<Integer> stream of [123]. 
```